.github/workflows/test.yml

```diff
@@ -626,3 +626,29 @@ jobs:
       - name: Test ${{ matrix.pkg }} Build
         run: |
           nix develop -c sh -c "cd pkg/${{ matrix.pkg }} ; zig build test"
+
+  test-debian-12:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: ["linux/amd64", "linux/arm64"]
+    name: Test build on Debian 12 ${{ matrix.platform }}
+    runs-on: namespace-profile-ghostty-sm
+    needs: test
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install and configure Namespace CLI
+        uses: namespacelabs/nscloud-setup@v0
+
+      - name: Configure Namespace powered Buildx
+        uses: namespacelabs/nscloud-setup-buildx-action@v0
+
+      - name: Build and push
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          file: src/build/docker/debian/Dockerfile
+          build-args: DISTRO_VERSION=12
```

Comment:
Let's pass the Zig version in here as well and put that as a top-level workflow env. The number of places we hardcode the Zig version we use is too damn high. 

Longer term it'd probably be better to make a `setup` job that extracts it from our source code (eventually the build.zig.zon but we don't use that yet).

Reply:
Fixed.